utils/get_product_by_id/index.js

```diff
@@ -0,0 +1,15 @@
+import axios from 'axios';
+import { SOP_PRODUCT_BY_ID_ENDPOINT } from '../../constants';
+
+export const get_product_by_id = async (id) => {
+  try {
+    const {
+      data: { Products },
+    } = await axios.get(`${SOP_PRODUCT_BY_ID_ENDPOINT}/${id}`);
```

Comment:
We can use axiosInstance over here. Since the base url will always remain same.
In utils we can have class that handles network request for the project and expose methods that will return api result. This will help in decoupling network calls from component.

Reply:
Sounds good , we can actually go 1 step further and make our own request handler function for this , that would work like a higher level api for multiple request types including handling of options building query string, i actually have worked on a similar kind of thing in my current project which works like a charm, let me know if you want me to go ahead with that , and we can discuss that further.

Reply:
Yes so it would be like wrapper function where we can give options as input and make network call.